src/dali/abstract_ccxt_input_plugin.py

```diff
@@ -147,7 +147,7 @@ def _get_markets(self) -> List[str]:
 
     @staticmethod
     def _rp2_timestamp_from_ms_epoch(epoch_timestamp: str) -> str:
-        rp2_time = datetime.fromtimestamp((int(epoch_timestamp) / _MS_IN_SECOND), timezone.utc)
+        rp2_time = datetime.fromtimestamp((float(epoch_timestamp) / _MS_IN_SECOND), timezone.utc)
 
         return rp2_time.strftime("%Y-%m-%d %H:%M:%S%z")
 
```

Comment:
You might as well just add another function for epochs in seconds (like what Kraken uses):
```
    @staticmethod
    def _rp2_timestamp_from_epoch(epoch_timestamp: str) -> str:
        rp2_time = datetime.fromtimestamp((float(epoch_timestamp)), timezone.utc)

        return rp2_time.strftime("%Y-%m-%d %H:%M:%S%z")
```

Then, you can use that in your code. I was just encountering a lot of epochs in milliseconds with the exchanges I was using.